src/social_edu_federation/backends/base.py

```diff
@@ -0,0 +1,111 @@
+    """
+    Wrapper around configuration for a SAML Identity provider.
+
+    This is the base class to all the SAMLIdentityProvider defined
+    in `social_edu_federation`.
+
+    For now it only provides a class method to easily instantiate
+    new objects from the metadata parsed dict.
+    """
+
+    @classmethod
+    def create_from_config_dict(cls, **kwargs):
+        """
+        Makes the conversion from the dictionary fetch by python3-saml
+        to an object usable by Python Social Auth.
+
+        Parameters
+        ----------
+        kwargs : dict
+            The identity provider we want to convert. Looks like:
+            ```
+            {
+                'name': 'idp-displayable-name',
+                'entityId': 'http://idp.domain/adfs/services/trust',
+                'singleSignOnService': {
+                    'url': 'https://idp.domain/adfs/ls/',
+                    'binding': 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect',
+                },
+                'singleLogoutService': {
+                    'url': 'https://idp.domain/adfs/ls/',
+                    'binding': 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect',
+                },
+                'x509cert': 'MIIC4DCCAcigAwIBAgIQG...CQZXu/agfMc/tY+miyrD0=',
+                'edu_fed_data': {
+                    'display_name': 'IdP displayable name',
+                    'organization_name': 'Organization',
+                    'organization_display_name': 'Organization displayable name',
+                },
+            }
+            ```
+
+        Returns
+        -------
+        Type[cls]
+            The object for Python Social Auth
+        """
+        edu_fed_parameters = kwargs.get("edu_fed_data", {})
+
+        return cls(
+            kwargs["name"],
+            entity_id=kwargs["entityId"],
+            url=kwargs.get("singleSignOnService", {}).get("url"),
+            slo_url=kwargs.get("singleLogoutService", {}).get("url"),
+            x509cert=kwargs.get("x509cert"),
+            x509certMulti=kwargs.get("x509certMulti"),
+            #
+            # Also store our own data in the class `conf` attribute
+            # edu_fed_display_name = edu_fed_parameters["display_name"]
+            # edu_fed_organization_display_name = edu_fed_parameters["organization_display_name"]
+            **{f"edu_fed_{key}": val for key, val in edu_fed_parameters.items()},
+        )
+
+
+class EduFedSAMLAuth(SAMLAuth):
+    """
+    This is the base class to all the SAMLAuth defined
+    in `social_edu_federation`.
+    """
+
+    # name must be overriden after (making it long to be painful for settings if forgotten)
+    name = "base_edu_fed_backend"
+    edu_fed_saml_idp_class = EduFedSAMLIdentityProvider
+
+    def get_federation_metadata_url(self):
+        """Boilerplate to the federation's metadata URL provided by settings."""
+        return self.setting("FEDERATION_SAML_METADATA", None)
```

Comment:
should the setting also be called a URL for clarify and coherence here?
```suggestion
        return self.setting("FEDERATION_SAML_METADATA_URL", None)
```